src/values.rs

```diff
@@ -411,16 +388,20 @@ impl JitValue {
                             let key = key.to_bytes_le();
                             let value = value.to_jit(arena, registry, &info.ty)?;
 
-                            let value_malloc_ptr =
-                                NonNull::new(libc::malloc(elem_layout.size())).unwrap();
+                            let value_malloc_ptr = libc::malloc(elem_layout.size());
 
                             std::ptr::copy_nonoverlapping(
                                 value.cast::<u8>().as_ptr(),
-                                value_malloc_ptr.cast().as_ptr(),
+                                value_malloc_ptr.cast(),
                                 elem_layout.size(),
                             );
 
-                            value_map.insert(key, value_malloc_ptr);
+                            value_map.insert(
+                                key,
+                                NonNull::new(value_malloc_ptr)
+                                    .expect("malloc shouldn't return null ptrs")
```

Comment:
This makes it sound like `malloc` is violating the contract when it's specified to return `NULL` on allocation failure. I'd rephrase the message as this: `"allocation failure"`. It is OK to `panic` since we tolerate that for Rust in general.